### ISSUE TITLE
Bayesian Neural Network Example - TypeError: Cannot convert value None to a TensorFlow DType.

### ISSUE THREAD

**Issue by image357** [tensorflow/probability#178]
Hey,

I get the following exception when I try to run the new Bayesian Neural Network example:
https://github.com/tensorflow/probability/blob/master/tensorflow_probability/examples/bayesian_neural_network.py

```python
TypeError                                 Traceback (most recent call last)
<ipython-input-1-babfab66cc96> in <module>
    307 
    308 if __name__ == "__main__":
--> 309     tf.app.run()

~\.virtualenvs\Tensorflow-yYwcsaOl\lib\site-packages\tensorflow\python\platform\app.py in run(main, argv)
    123   # Call the main function, passing through any arguments
    124   # to the final program.
--> 125   _sys.exit(main(argv))
    126 

<ipython-input-1-babfab66cc96> in main(***failed resolving arguments***)
    224         ])
    225 
--> 226     logits = neural_net(images)
    227     labels_distribution = tfd.Categorical(logits=logits)
    228 

~\.virtualenvs\Tensorflow-yYwcsaOl\lib\site-packages\tensorflow\python\keras\engine\base_layer.py in __call__(self, inputs, *args, **kwargs)
    757           # the user has manually overwritten the build method do we need to
    758           # build it.
--> 759           self.build(input_shapes)
    760         # We must set self.built since user defined build functions are not
    761         # constrained to set self.built.

~\.virtualenvs\Tensorflow-yYwcsaOl\lib\site-packages\tensorflow\python\keras\engine\sequential.py in build(self, input_shape)
    220         if not layer.built:
    221           with ops.name_scope(layer._name_scope()):
--> 222             layer.build(shape)
    223           layer.built = True
    224         shape = layer.compute_output_shape(shape)

~\.virtualenvs\Tensorflow-yYwcsaOl\lib\site-packages\tensorflow_probability\python\layers\conv_variational.py in build(self, input_shape)
    199     input_dim = input_shape[channel_axis].value
    200     kernel_shape = self.kernel_size + (input_dim, self.filters)
--> 201     dtype = tf.as_dtype(self.dtype)
    202 
    203     # Must have a posterior kernel.

~\.virtualenvs\Tensorflow-yYwcsaOl\lib\site-packages\tensorflow\python\framework\dtypes.py in as_dtype(type_value)
    710             type_value, e))
    711 
--> 712   raise TypeError("Cannot convert value %r to a TensorFlow DType." % type_value)

TypeError: Cannot convert value None to a TensorFlow DType.
```

My pip freeze:
```
absl-py==0.5.0
astor==0.7.1
backcall==0.1.0
colorama==0.3.9
cycler==0.10.0
decorator==4.3.0
gast==0.2.0
grpcio==1.15.0
h5py==2.8.0
ipykernel==5.0.0
ipython==7.0.1
ipython-genutils==0.2.0
jedi==0.13.1
jupyter-client==5.2.3
jupyter-core==4.4.0
Keras-Applications==1.0.6
Keras-Preprocessing==1.0.5
kiwisolver==1.0.1
Markdown==3.0.1
matplotlib==3.0.0
numpy==1.15.2
pandas==0.23.4
parso==0.3.1
pickleshare==0.7.5
prompt-toolkit==2.0.5
protobuf==3.6.1
Pygments==2.2.0
pyparsing==2.2.2
python-dateutil==2.7.3
pytz==2018.5
pyzmq==17.1.2
scipy==1.1.0
simplegeneric==0.8.1
six==1.11.0
tensorboard==1.11.0
tensorflow==1.11.0
tensorflow-probability==0.3.0
termcolor==1.1.0
tornado==5.1.1
traitlets==4.3.2
wcwidth==0.1.7
Werkzeug==0.14.1
```

**Comment by csuter** [comment]
Hi @image357 -- looks like you had TF 1.11 with TF Probability 0.3; the latter was built and tested against TF 1.10.

We just released TFP 0.4, which will work with TF 1.11 -- can you upgrade your TFP install and see if that fixes the issue?

**Comment by image357** [comment]
It is fixed with TFP 0.4. Thanks!
